src/materialized/src/bin/materialized/tracing.rs

```diff
@@ -124,6 +126,8 @@ where
 /// incrementing a counter for the severity of messages reported.
 pub struct MetricsRecorderLayer<S> {
     counter: ThirdPartyMetric<IntCounterVec>,
+    level_counter:
+        Arc<RwLock<HashMap<String, DeleteOnDropCounter<'static, AtomicI64, Vec<String>>>>>,
```

Comment:
Hmmmm, why is this turning into delete-on-drop? I think this lives as long as the server does (and I'm not loving the look of the locked hashmap)

Reply:
More majorly, this change removes the `ThirdPartyMetric`, which will break cloud once we actually attempt to read that metric /:

Reply:
Reverted! `ThirdPartyMetrics` should be unaffected by this change now. Clients might have to pass the raw variant of a metric instead of the d-o-d wrapped metric.